docs/rules/no-namespace-imports.md

```diff
@@ -0,0 +1,46 @@
+# Prevent namespace import declarations. (no-namespace-imports)
+
+Namespace imports grab all exported variables from a module as nested properties under a single variable name. This approach of importing everything, rather than only what is required from the module, leads to a polluted local scope from the unused components that were brought in under the module namespace.
+
+## Rule Details
+
+This rule will report all namespace imports and will optionally fix them (with the `--fix` flag) by converting them to default imports.
+
+Examples of **incorrect** code for this rule:
+
+```ts
+
+import * as React from 'react';
+
+```
+
+Examples of **correct** code for this rule:
+
+```ts
+
+import React from 'react';
```

Comment:
I'd rather show an example showing named imports > namespace import. Yes, this rule does help us enforce the right style for React, but that's something that could conceivably be addressed in types instead.